lib/ecto/adapters/postgres/connection.ex

```diff
@@ -509,6 +526,25 @@ if Code.ensure_loaded?(Postgrex) do
       [aggregate, " FILTER (WHERE ", expr(filter, sources, query), ?)]
     end
 
+    defp expr({:over, _, [agg, %PartitionByExpr{} = window]}, sources, query) do
+      aggregate = expr(agg, sources, query)
+      [aggregate, " OVER ", partition_by(window, sources, query)]
+    end
+
+    defp expr({:over, _, [agg, nil]}, sources, query) do
+      aggregate = expr(agg, sources, query)
+      [aggregate, " OVER ()"]
+    end
+
+    defp expr({:over, _, [agg, name]}, sources, %Ecto.Query{ windows: windows } = query) when is_atom(name) do
+      if Keyword.has_key?(windows, name) do
```

Comment:
This validation should happen in Ecto.Query.Planner. The adapter should always assume that query is correct (unless it doesn't support a particular construct).